File: archives/Installer/Files/Templates/LivetMessage_CSharp/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;

using Livet.Messaging;

namespace $rootnamespace$
{
    /*
     * 戻り値のある相互作用メッセージはResponsiveInteractionMessage<T>を継承して作成します。
     * Tは戻り値の型です。
     * 戻り値のない相互作用メッセージはInteractionMessageを継承して作成します。
     */
    public class $safeitemname$ : ResponsiveInteractionMessage<string>
    {
        //Viewでメッセージインスタンスを生成する時のためのコンストラクタ
        public $safeitemname$()
        {
        }

        //ViewModelからMessenger経由での発信目的でメッセージインスタンスを生成するためのコンストラクタ
        public $safeitemname$(string messageKey)
            : base(messageKey)
        {

        }

        /*
         * メッセージに保持させたい情報をプロパティとして定義してください。
         * Viewでバインド可能なプロパティにするために依存関係プロパティとして定義する事をお勧めします。
         * 通常依存関係プロパティはコードスニペット propdpを使用して定義します。
         * もし普通のプロパティとして定義したい場合はコードスニペット propを使用して定義します。
         */

        /// <summary>
        /// 派生クラスでは必ずオーバーライドしてください。Freezableオブジェクトとして必要な実装です。<br/>
        /// 通常このメソッドは、自身の新しいインスタンスを返すように実装します。
        /// </summary>
        /// <returns>自身の新しいインスタンス</returns>
        protected override System.Windows.Freezable CreateInstanceCore()
        {
            return new $safeitemname$();
        }
    }
}

File: archives/Installer/Files/Templates/LivetMessage_VB/Message.vb
'戻り値のある相互作用メッセージはResponsiveInteractionMessage(Of T)を継承して作成します。
'Tは戻り値の型です。
'戻り値のない相互作用メッセージはInteractionMessageを継承して作成します。
Public Class $safeitemname$
    Inherits ResponsiveInteractionMessage(Of String)

    'Viewでメッセージインスタンスを生成する時のためのコンストラクタ
    Public Sub New()

    End Sub

    'ViewModelからMessenger経由での発信目的でメッセージインスタンスを生成するためのコンストラクタ
    Public Sub New(ByVal messageKey As String)
        MyBase.New(messageKey)
    End Sub
    
    'メッセージに保持させたい情報をプロパティとして定義してください。
    'Viewでバインド可能なプロパティにするために依存関係プロパティとして定義する事をお勧めします。
    '通常依存関係プロパティはコードスニペット propdpを使用して定義します。
    'もし普通のプロパティとして定義したい場合はコードスニペット propを使用して定義します。
    
    ''' <summary>
    ''' 派生クラスでは必ずオーバーライドしてください。Freezableオブジェクトとして必要な実装です。<br/>
    ''' 通常このメソッドは、自身の新しいインスタンスを返すように実装します。
    ''' </summary>
    ''' <returns>自身の新しいインスタンス</returns>
    Protected Overrides Function CreateInstanceCore() As System.Windows.Freezable
        Return New $safeitemname$()
    End Function
End Class